src/bscPlugin/validation/ScopeValidator.ts

```diff
@@ -523,29 +523,76 @@ export class ScopeValidator {
             return;
         }
 
-        if (!isComponentType(callerType)) {
-            this.addMultiScopeDiagnostic({
-                ...DiagnosticMessages.cannotFindCallFuncFunction(methodName, functionFullname, callerType.toString()),
-                location: callErrorLocation
-            });
-            return;
-        }
-
         const funcType = util.getCallFuncType(expression, methodToken, { flags: SymbolTypeFlag.runtime, ignoreCall: true });
         if (!funcType?.isResolvable()) {
             this.addMultiScopeDiagnostic({
                 ...DiagnosticMessages.cannotFindCallFuncFunction(methodName, functionFullname, callerType.toString()),
                 location: callErrorLocation
             });
         }
-        return this.validateFunctionCall(file, funcType, callErrorLocation, expression.args);
+
+        return this.validateFunctionCall(file, expression, funcType, callErrorLocation, expression.args);
     }
 
     /**
      * Detect calls to functions with the incorrect number of parameters, or wrong types of arguments
      */
-    private validateFunctionCall(file: BrsFile, funcType: BscType, callErrorLocation: Location, args: Expression[], argOffset = 0) {
-        if (!funcType?.isResolvable() || !isTypedFunctionType(funcType)) {
+    private validateFunctionCall(file: BrsFile, callee: Expression, funcType: BscType, callErrorLocation: Location, args: Expression[], argOffset = 0) {
+        if (!funcType?.isResolvable() || !isCallableType(funcType)) {
+            const funcName = util.getAllDottedGetPartsAsString(callee, ParseMode.BrighterScript, isCallfuncExpression(callee) ? '@.' : '.');
+            if (isUnionType(funcType)) {
```

Comment:
Extra checking for unions of functions... we try to boild it down to a function type that is checkable